src/app/pages/timeTable/timeTable.component.ts

```diff
@@ -0,0 +1,191 @@
+        this.startDate.setHours(0, 0, 0, 0);
+        this.endDate.setHours(23, 59, 59, 999);
+        this.doctorService.getAll().subscribe(doctors => {
+            let doctorTimeSlots: Observable<[Doctor, TimeSlot[]]>[] = doctors.map(doctor => {
+                let timeSlots: Observable<TimeSlot[]> = this.timeSlotService
+                    .getTimeSlots(doctor, this.startDate, this.endDate);
+                return timeSlots.map(slot => [doctor, slot])
+            })
+            Observable.forkJoin(doctorTimeSlots).subscribe((everyDoctorData: [Doctor, TimeSlot[]][]) =>
+                this.refreshWithDataReady(everyDoctorData))
+        })
+    }
+
+    private refreshWithDataReady(everyDoctorData: [Doctor, TimeSlot[]][]): void {
+        var newDoctorInfos: DoctorInfo[] = []
+        var newDatesSequence: string[] = []
+        let daysCount = Math.round((this.endDate.getTime() - this.startDate.getTime()) / 1000 / 60 / 60 / 24)
+        for (var singleDoctorData of everyDoctorData) {
+            var [doctor, slots] = singleDoctorData
+            this.processSingleDoctor(doctor, slots, newDoctorInfos, newDatesSequence, daysCount)
+        }
+        this.doctorInfos = this.sortSlotsInsideDoctorInfo(newDoctorInfos)
+        this.datesSequence = newDatesSequence
+    }
+
+    private processSingleDoctor(doctor: Doctor, slots: TimeSlot[],
+        doctorInfos: DoctorInfo[], datesSequence: string[], daysCount: number): void {
+        var processedDocInfo: DoctorInfo = this.fetchDoctorToProcess(doctor, doctorInfos, daysCount)
+        for (var dayOffset = 0; dayOffset < daysCount; dayOffset++) {
+            let dayStartTime: Date = new Date(this.startDate.getTime())
+            dayStartTime.setDate(this.startDate.getDate() + dayOffset)
+            let dayEndTime: Date = new Date(this.startDate.getTime())
+            dayEndTime.setDate(this.startDate.getDate() + dayOffset + 1)
+            datesSequence.push(dayStartTime.getDate() + "." + (dayStartTime.getMonth() + 1))
+            let slotInfos: SlotInfo[] = this.createSlotInfosForDay(slots, dayStartTime, dayEndTime)
+            var resultArray: SlotInfo[] = processedDocInfo.subsequentDaysInfo[dayOffset]
+            for (var slot of slotInfos) {
+                resultArray.push(slot)
+            }
+        }
+    }
+
+    private createSlotInfosForDay(slots: TimeSlot[], dayStartTime: Date, dayEndTime: Date): SlotInfo[] {
+        return slots.filter(slot =>
+            slot.startDateTime.getTime() > dayStartTime.getTime()
+            && slot.endDateTime.getTime() < dayEndTime.getTime())
+            .map(slot => {
+                return {
+                    startTime: slot.startDateTime,
+                    endTime: slot.endDateTime,
+                    asString: this.toHourString(slot.startDateTime, slot.endDateTime)
+                }
+            })
+    }
+
+    private fetchDoctorToProcess(doctor: Doctor, doctorInfos: DoctorInfo[], daysCount: number): DoctorInfo {
```

Comment:
Lepszą nazwą byłoby `getOrCreateDoctorInfo` (bo jak rozumiem to właśnie ta metoda robi)